### ISSUE TITLE
(v2) Multi-language support (starting with German/DE)

### ISSUE THREAD

**Issue by proddy** [proddy/EMS-ESP#426]
Support multi-locale for Telnet console commands, MQTT, SysLog messages and the Web. Since there is not enough memory to hold all the translations it will have to be different builds (DE, NL, UK). String literals will be stored in translation files and built at compile time.

@MichaelDvP @bbqkees what you vote for this feature?

**Comment by bbqkees** [comment]
Most Gateways go to German speaking countries so it makes perfect sense to provide the interface in their own language. I get some questions about it regularly.
If there is not enough memory to store everything, we should at least provide a language selector at the first install which will then download and install the new firmware. 
So we could just have the UK build as default with a selector to install another language.

**Comment by MichaelDvP** [comment]
Yes, that would be nice. What about a changeable language-file in the spiffs? There is a lot of space left and you have only one software.

**Comment by proddy** [comment]
That's a good idea. Store the localizations in SPIFFS and download to PROGMEM when EMS-ESP boots.

**Comment by proddy** [comment]
In order of importance
 - Web UI
 - MQTT topics and payload
 - Console commands (serial and telnet)
 - Logging information (to console & syslog)

For the web I had a look at react-intl, i18next-react, polyglot and this one called lingui-js but they'll all quite big libraries and there is just not enough progmem to store all the translations to have it real-time on the ESP8266. react-intl is the right one here (14k zipped).

**Comment by proddy** [comment]
First localized version of the UI is looking promising.

**Comment by proddy** [comment]
with the extra 160K on the ESP32 I can easily store the strings in SPIFFS and load them dynamically
